src/common/debug.c

```diff
@@ -238,11 +237,16 @@ daos_key2str(daos_key_t *key)
 				break;
 			}
 		}
-		if (can_print) {
-			strncpy(buf, key->iov_buf, len);
-			buf[len] = 0;
-		} else {
-			strcpy(buf, "????");
+
+		if (can_print)
+			snprintf(buf, len, "%s", (char *)key->iov_buf);
+		else
+			sprintf(buf, "????");
+
+		if (key->iov_len == sizeof(uint64_t)) {
```

Comment:
Are you anticipating the case that a buffer is both a valid string, and you want to print it as a base 10 number which is what this could would appear to do?

Would it not make more sense to do a hex dump of the iov, up to the maximum of iov_len or  DF_KEY_STR_SIZE, allowing for ellipses if the key is longer.

Reply:
>Are you anticipating the case that a buffer is both a valid string, and you want to print it as a base 10 number which is what this could would appear to do?

Yes, even with base 10 number, 32 bytes would be enough for string "uint64:xxxx", well, dump individual hex bytes is not friendly for debugging, which is the point of adding this extra information.